handsontable/src/plugins/columnSorting/__tests__/columnSorting.spec.js

```diff
@@ -1018,19 +1018,19 @@ describe('ColumnSorting', () => {
 
       getPlugin('columnSorting').sort({ column: 2, sortOrder: 'asc' }); // ASC
 
-      expect(getDataAtRow(0)).toEqual(['Mercedes', 'A 160', '01/14/2006', 6999.9999]);
-      expect(getDataAtRow(1)).toEqual(['Opel', 'Astra', '02/02/2004', 7000]);
-      expect(getDataAtRow(2)).toEqual(['BMW', '320i Coupe', '07/24/2011', 30500]);
-      expect(getDataAtRow(3)).toEqual(['Audi', 'A4 Avant', '11/19/2011', 33900]);
-      expect(getDataAtRow(4)).toEqual(['Citroen', 'C4 Coupe', '12/01/2008', 8330]);
+      expect(getDataAtRow(0)).toEqual(['Opel', 'Astra', '02/02/2004', 7000]);
```

Comment:
@budnix I need some root-cause analysis of why is this changed.

Reply:
It seems that before updating the lib, the date values sorting does not work correctly. At least in that test case. For example, the results for sorting rows ascending:
Before (it's sorted without any noticeable pattern):
```js
'01/14/2006'
'02/02/2004'
'07/24/2011'
'11/19/2011'
'12/01/2008'
``` 
After:
```js
'02/02/2004'
'01/14/2006'
'12/01/2008'
'07/24/2011'
'11/19/2011'
```

Reply:
I am sorry, but this is not the root cause. The root cause seems related to the fact that the test, despite its name, provided *wrongly formatted* data to Moment.js, and tested that Moment.js *does not* sort in that case. After the change, Moment.js sorts even when it is given wrongly formatted data.

Compare the format given in line 1009 and the data provided in lines 996-1002

https://github.com/handsontable/handsontable/blob/5ec0fa6500d6b20d941e79aee4d53d0a940b8885/handsontable/src/plugins/columnSorting/__tests__/columnSorting.spec.js#L993-L1009

Reply:
Maybe we should use Moment.js in the strict mode? https://momentjs.com/guides/#/parsing/strict-mode/

Reply:
> Maybe we should use Moment.js in the strict mode? https://momentjs.com/guides/#/parsing/strict-mode/

I would be for it. It's better to only sort the date-type values based on the `dateFormat`. Without leaving the room for guesses.

Reply:
A little root cause analysis by me:

- in the [forgiving mode](https://momentjs.com/guides/#/parsing/forgiving-mode/), if Moment cannot parse a date, it uses `Date` constructor
- in Moment 2.24.0, the date `moment('01/14/2006', 'mm/dd/yy')` was not parsed at all. The `Date` constructor fallback returned the **current datetime** (https://jsfiddle.net/warpech/d9qako0g/4/)
- in Moment 2.25.0 (actually in this commit https://github.com/moment/moment/commit/0c7d8da2b7563363ef7e3de2b9a7023090d0462f), the character `yy` became significant and that date was parsed without resolving to the `Date` constructor. However, parsing is not good either. Moment parses this date to `2006-01-01`. (https://jsfiddle.net/warpech/d9qako0g/4/)
- with `strict` flag turned off, all tested Moment versions from 2.24.0 to 2.29.2 consistently say it is an incorrect date.

Reply:
I was advised that there are parts of Handsontable that treat the forgiving mode as a feature and we should not remove it. Discussion: https://handsoncode.slack.com/archives/C4PV4S33Q/p1650384660407479?thread_ts=1649328227.856499&cid=C4PV4S33Q